src/index.js

```diff
@@ -1093,6 +1071,50 @@ export class ULabel {
         return ULabel.version();
     }
 
+    /**
+     * Find all toolbox items that contain overlays, add a reference to them, and add them to the document
+     * Currently only FilterDistance has an overlay to check for
+     */
+    create_overlays() {
+        // Create an array that states which ToolboxItems want to create an overlay. Currently only one, but may be expanded
+        const possible_overlays = [
+            "FilterDistance"
+        ]
+
+        for (const toolbox_item of this.toolbox.items) {
+            // Store current toolbox name in a constant for convenience
+            const toolbox_name = toolbox_item.get_toolbox_item_type()
+
+            // If the current toolboxitem is not included in possible_overlays then continue
+            if (!possible_overlays.includes(toolbox_name)) continue
+
+            switch(toolbox_name) {
+                case "FilterDistance":
+                    // Give ulabel a referance to the filter overlay for confinience
+                    this.filter_distance_overlay = toolbox_item.get_overlay()
```

Comment:
so if additional overlays are added in the future, they'll all need their own custom update logic?

Reply:
Currently yes. For performance reasons I wanted all of the rendering logic to be separated out. That way overlays will only be re-rendered when they need to, instead of when any overlay needs to. 

If we do end up making a few more overlays it'd be cool if we made an OverlayHandler class or something similar that could tie into ULabel's state and handle all of the update logic, instead of baking it into the ULabel class.